metricbeat/mb/testing/testdata.go

```diff
@@ -362,7 +362,11 @@ func writeDataJSON(t *testing.T, data mapstr.M, path string) {
 	// Add hardcoded timestamp
 	data.Put("@timestamp", "2019-03-01T08:05:34.853Z")
 	output, err := json.MarshalIndent(&data, "", "    ")
-	if err = ioutil.WriteFile(path, output, 0644); err != nil {
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
```

Comment:
Same here
```suggestion
		t.Fatal(err)
```

Reply:
Looking at existing code, `t.Error(err)` followed by `t.FailNow()` is the most common pattern. So far, no one cared that the error was ignored so far. And the change was added, just to make linter happy.

Reply:
> Looking at existing code, t.Error(err) followed by t.FailNow() is the most common pattern.

The bits of the codebase I mostly work with follow a different pattern 😅 , anyway this one is not a blocker.